### ISSUE TITLE
Move react as peerDependency

### ISSUE THREAD

**Issue by jabranr** [xiaolin/react-image-gallery#431]
First of all great work on this package.

I have come across an issue of multiple `react` installations. This was not a problem during development and bundling for production but more in testing. The tests just fail when run in combination with Create React App with an error saying that there may be more than once installations of `react`. I did `npm ls react` at root level and found that is the case:

```
components@1.0.0 components
├─┬ media-gallery@1.0.1 -> components/src/packages/media-gallery
│ └─┬ react-image-gallery@0.8.18
│   └── react@16.9.0
└── react@16.9.0
```

I think moving `react` to be as a `peerDependency` would resolve this or similar problems. This obviously be a breaking change so would require a major release in case it goes ahead.

Let me know what you think. Cheers

**Comment by xiaolin** [comment]
If react is removed from the dependency list, this package won't work correctly in standalone. There is probably a way to modify your test so that it does not install duplicate dependencies. Maybe `npm dedupe` (https://docs.npmjs.com/cli/dedupe.html) can help.

**Comment by jabranr** [comment]
I did not mean to remove `react` as dependency. Only requesting it to be listed as `peerDependency`. It will still be required but need to be installed by the consumer of this package. Basically following will achieve it:

```
npm install react react-image-gallery
```

**Comment by xiaolin** [comment]
Ah, hmm. I've not added the same package as `peerDependecy` and `dependecy`. Any documentation on this or know the type of implications it may cause?

**Comment by jabranr** [comment]
Sure. Here are few examples of it and some quick reading.

- https://nodejs.org/es/blog/npm/peer-dependencies/
- https://stackoverflow.com/a/34645112/428901
- https://github.com/KyleAMathews/typography.js/blob/master/packages/react-typography/package.json#L7

If making a PR makes it easier to understand the concept then I am happy to quickly put one through.

**Comment by xiaolin** [comment]
@jabranr Thanks, I reviewed your pr and the changes shouldn't be breaking. If you can tweak the pr I can merge.

**Comment by xiaolin** [comment]
released in v0.9.0
